core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -439,7 +439,14 @@ private[spark] object JsonProtocol {
     val reason = Utils.getFormattedClassName(taskEndReason)
     val json: JObject = taskEndReason match {
       case fetchFailed: FetchFailed =>
-        val blockManagerAddress = Option(fetchFailed.bmAddress).
+        val blockManagerAddress = Option {
+          fetchFailed.bmAddress match {
+            case bmId: BlockManagerId =>
+              bmId
+            case _ =>
+              // TODO: support custom locations
```

Comment:
`IDed TODO` is always better for sharing this work item with the Apache Spark community.